### DISCUSSION TITLE
[Specs] ScrollAware revealer (Flutter, Facebook, toolbar and bottom bar hide on scroll)

### DISCUSSION THREAD

**Discussion by roubachof** [unoplatform/uno.toolkit.ui#852] [Ideas]
Would be great to have a simple away to implement the most used "hide on scroll" UX pattern.
2 simple attached properties should be enough:

| Name                  | Type             | Description                                                |
| --------------------- | ---------------- | ---------------------------------------------------------- |
| ScrollViewer        | `ScrollViewer`         | the ScrollViewer controlling the reveal effect                              |
| RevealedPosition| `Enumeration` | Enum: Left, Top, Bottom, Right          |

```xaml
 <Grid>

    <ScrollViewer x:Name="ScrollViewer">
        <StackPanel>
            <Item />
            <Item />
            <Item />
			...
        </StackPanel>
    </ScrollViewer>

	<StackPanel Padding="20"
	            VerticalAlignment="Top"
	            local:ScrollAware.RevealedPosition="Top"
	            local:ScrollAware.ScrollViewer="{x:Bind ScrollViewer}"
	            Background="#333333">
	    ...
	</StackPanel>

	<StackPanel Width="100"
	            Height="300"
	            HorizontalAlignment="Left"
	            VerticalAlignment="Center"
	            local:ScrollAware.RevealedPosition="Left"
	            local:ScrollAware.ScrollViewer="{x:Bind ScrollViewer}"
	            Background="#333333"
	            CornerRadius="0,30,30,0">
	    ...
	</StackPanel>

	<StackPanel Width="100"
	            Height="300"
	            HorizontalAlignment="Right"
	            VerticalAlignment="Center"
	            local:ScrollAware.RevealedPosition="Right"
	            local:ScrollAware.ScrollViewer="{x:Bind ScrollViewer}"
	            Background="#333333"
	            CornerRadius="30,0,0,30">
	    ...
	</StackPanel>
	
	<StackPanel Padding="20"
	            VerticalAlignment="Bottom"
	            local:ScrollAware.RevealedPosition="Bottom"
	            local:ScrollAware.ScrollViewer="{x:Bind ScrollViewer}"
	            Background="#333333">
	    ...
	</StackPanel>

</Grid>
```

https://github.com/unoplatform/uno.toolkit.ui/assets/596903/344392b1-0d12-418b-b09b-6e7067c9272c

In the current implementation there is an abstract `ScrollAwareController` that is used to subscribe to scroll events and provides the need properties such as velocity, offset delta and vertical offset.

This could be used to implement https://github.com/unoplatform/uno.toolkit.ui/discussions/768

I already implemented a working component here: https://github.com/sharpnado-pro/Uno.PoC
@kazo0 and @pictos are admin, you can request them the read permission.
You call also see the whole scroll aware code here: https://gist.github.com/roubachof/d77099ededf9e75d47453996a3250c75

https://github.com/unoplatform/uno.toolkit.ui/assets/596903/01182657-9426-48ca-8526-6fe415b6698c

**Comment by Xiaoy312** [comment]
we could maybe add a property to adjust the triggering scroll direction? which should be defaulted to scroll up
